webapp/channels/src/components/widgets/advanced_textbox/advanced_textbox.tsx

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import React, {useState, useEffect} from 'react';
+
+import ShowFormat from 'components/advanced_text_editor/show_formatting/show_formatting';
+import Textbox from 'components/textbox';
+import type {TextboxElement} from 'components/textbox';
+import type TextboxClass from 'components/textbox/textbox';
+
+import './advanced_textbox.scss';
+
+type AdvancedTextboxProps = {
+    id: string;
+    value: string;
+    channelId: string;
+    onChange: (e: React.ChangeEvent<TextboxElement>) => void;
+    onKeypress: (e: React.KeyboardEvent<TextboxElement>) => void;
+    createMessage: string;
+    characterLimit: number;
+    minCharacterLimit?: number;
+    minLengthErrorMessage?: string;
+    preview: boolean;
+    togglePreview: () => void;
+    textboxRef?: React.RefObject<TextboxClass>;
+    useChannelMentions?: boolean;
+    descriptionMessage?: JSX.Element | string;
+    hasError?: boolean;
+    errorMessage?: string | JSX.Element;
+    onValidate?: (value: string) => { isValid: boolean; errorMessage?: string };
+    showCharacterCount?: boolean;
+    readOnly?: boolean;
+};
+
+const AdvancedTextbox = ({
+    id,
+    value,
+    channelId,
+    onChange,
+    onKeypress,
+    createMessage,
+    characterLimit,
+    minCharacterLimit,
+    minLengthErrorMessage,
+    preview,
+    togglePreview,
+    textboxRef,
+    useChannelMentions = false,
+    descriptionMessage,
+    hasError,
+    errorMessage,
+    onValidate,
+    showCharacterCount = false,
+    readOnly = false,
+}: AdvancedTextboxProps) => {
+    const [internalError, setInternalError] = useState<string | JSX.Element | undefined>(errorMessage);
```

Comment:
I'm not sure I understand why `errorMessage` is passed in as a prop and then used for this. I'd expect the error state to come from entirely outside this component or be only internal